dungeon/src/task/quizquestion/Quiz.java

```diff
@@ -0,0 +1,149 @@
+     *     image, or both text and a path to an image?
+     * @param questionText The question itself (can contain a path to images).
+     */
+    public Quiz(final Type type, Content.Type questionContentType, String questionText) {
+        super();
+        this.type = type;
+        taskText(questionText);
+        question = new Content(this, questionContentType, questionText);
+    }
+
+    /**
+     * Get the type of the question.
+     *
+     * @return The type of the question.
+     */
+    public Type type() {
+        return type;
+    }
+
+    /**
+     * Create a {@link Content} answer instance with the given configuration as a possible answer
+     * for this question.
+     *
+     * @param type What does the answer contain? Just text, just a path to an image, or both text
+     *     and a path to an image?
+     * @param content The answer itself (can contain a path to images).
+     */
+    public void addAnswer(Content.Type type, String content) {
+        addContent(new Content(this, type, content));
+    }
+
+    /**
+     * Get the question instance.
+     *
+     * @return The question instance.
+     */
+    public Content question() {
+        return question;
+    }
+
+    /**
+     * The QuizQuestionType enum represents the different types of quiz questions that can be
+     * created. The available types are SINGLE_CHOICE, MULTIPLE_CHOICE, and FREETEXT. SINGLE_CHOICE
+     * represents a question with multiple answer choices, where the user is required to select one
+     * answer. MULTIPLE_CHOICE represents a question with multiple answer choices, where the user is
+     * allowed to select multiple answers. FREETEXT represents a question where the user is required
+     * to input their answer as text.
+     */
+    public enum Type {
+        SINGLE_CHOICE,
+        MULTIPLE_CHOICE,
+        FREETEXT
+    }
+
+    /**
+     * Content for a {@link Quiz}-
+     *
+     * <p>Is used as answer and question for a {@link Quiz}.
+     *
+     * <p>Stores a String with the question/answer text and {@link Type} which defines if the String
+     * is just Text or contains a path to an image or both. The type is used by the {@link QuizUI}
+     * to configure the ui if the question is asked via the ui.
+     */
+    public static class Content extends TaskContent {
+
+        private final Type type;
+        private final String content;
+
+        /**
+         * Creates a new {@link Content}.
+         *
+         * <p>Use {@link Content#addAnswer(Type, String)} to create an instance of this.
+         *
+         * @param task Task to which this content belongs.
+         * @param type What does the answer contain? Just text, just a path to an image, or both
+         *     text and a path to an image?
+         * @param content The answer itself (can contain a path to images).
+         */
+        private Content(Quiz task, Type type, String content) {
```

Comment:
finde ich besser jetzt.

noch zwei gedanken:

(a) sollte es möglich sein, den content quasi von außen zusammenzuschrauben (also ohne die bindung an einen task) und erst beim hinzufügen zu einem task das objekt dann daran zu binden? sonst sind die operationen irgendwie doppelt: der content-ctor will den task haben, und dann muss ich auf dem task nochmal extra "add" für den content aufrufen?

(b) muss ich den typ wirklich im ctor übergeben? wir haben drei content-typen, und jeder typ hat seine spezifischen merkmale, die ich im ctor setzen muss. wenn ich also den ctor mit nur nem string aufrufe, dann meine ich implizit `text` (sonst würde ich ein bild oder beides übergeben).

Reply:
b) Wir hatten uns irgendwann mal darauf festgelegt, dass im String sowohl der Text als auch der Pfad zum Bild steht. Also irgendwas wie "Guck mal hier /meine/tolle/katze.png Ist die nicht süß?" Das macht die eingabe in der DSL für die Lehrperson deutlich einfacher. Da so im String ganz bequem der Pfad und die "stelle" im Text an der das Bild einzufügen ist, angegeben werden kann. 
Das Ui muss allerdings wissen was da zu sehen ist, daher der Typ.
=> Ja der Typ sollte im ctor angegeben werden.
Man könnte den String noch dynamisch analysieren (also gucken ob da irgendwo ein Pfad drinne ist), aber das macht irgendwie nur Probleme und impliziert wieder krams.

Reply:
hmmm? das klingt irgendwie nach blödsinn?